tests/compile-fail/mut_mut.rs

```diff
@@ -30,14 +28,37 @@ fn main() {
     }
 
     if fun(x) {
+        let y : &mut &mut u32 = &mut &mut 2;
+        //~^ ERROR generally you want to avoid `&mut &mut
+        //~| ERROR generally you want to avoid `&mut &mut
+        //~| ERROR generally you want to avoid `&mut &mut
+        **y + **x;
+    }
+
+    if fun(x) {
         let y : &mut &mut &mut u32 = &mut &mut &mut 2;
         //~^ ERROR generally you want to avoid `&mut &mut
         //~| ERROR generally you want to avoid `&mut &mut
         //~| ERROR generally you want to avoid `&mut &mut
         //~| ERROR generally you want to avoid `&mut &mut
+        //~| ERROR generally you want to avoid `&mut &mut
+        //~| ERROR generally you want to avoid `&mut &mut
```

Comment:
The PR fixes an actual annoying bug with `for _ in &mut bar` but in this case also duplicates some errors. On the other hand `&mut &mut &mut` is such a corner case I did not think it would be worth it to try to deduplicate errors here.